cli/dashboard_device_commands.go

```diff
@@ -4,143 +4,304 @@
+		}
 
-				if ctx.Args().Len() < 2 {
-					newName = interactiveTextInput("New device name")
-				} else {
-					newName = ctx.Args().Get(1)
-				}
+		resp, err := requests.UpdateDevice(uuid, params)
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
 
-				callDashboardAPI(func(client graphql.Client) (*generated.RenameDeviceResponse, error) {
-					return generated.RenameDevice(client, deviceIp, newName)
-				})
+		if rawJson {
+			output.PrintJsonOrError(resp)
+		} else {
+			prettyPrintDevice(resp)
+		}
+		return nil
+	},
+}
 
-				printSuccess("Device renamed successfully.")
+var dashboardDeviceUnclaimCommand = &cli.Command{
+	Name:      "unclaim",
+	Usage:     "Unclaim self (no argument) or some other device (single argument) in your network",
+	ArgsUsage: "<device uuid OR Husarnet IPv6 addr OR hostname>",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:  "yes",
+			Usage: "don't ask for confirmation",
+		},
+	},
+	Action: func(ctx context.Context, cmd *cli.Command) error {
+		requiredArgumentsRange(cmd, 0, 1)
 
+		if cmd.Args().Len() == 0 {
+			if !cmd.Bool("yes") {
+				askForConfirmation("Are you sure you want to unclaim this device?")
+			}
+			resp, err := requests.UnclaimSelf()
+			if err != nil {
+				printError(err.Error())
 				return nil
-			},
-		},
-		{
-			Name:      "remove",
-			Aliases:   []string{"rm"},
-			Usage:     "remove device from your account",
-			ArgsUsage: "[device ip]",
-			Action: func(ctx *cli.Context) error {
-				requiredArgumentsRange(ctx, 0, 1)
-
-				var deviceIp string
-
-				if ctx.Args().Len() < 1 {
-					deviceIp = interactiveGetDeviceIpByName()
-				} else {
-					deviceIp = getDeviceIpByNameOrIp(ctx.Args().Get(0))
-				}
+			}
 
-				callDashboardAPI(func(client graphql.Client) (*generated.RemoveDeviceResponse, error) {
-					return generated.RemoveDevice(client, deviceIp)
-				})
+			if rawJson {
+				output.PrintJsonOrError(resp)
+			}
+			return nil
+		}
 
-				printSuccess("Device removed successfully.")
+		uuid, err := determineDeviceUuid(cmd.Args().First())
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
 
-				return nil
-			}},
+		resp, err := requests.UnclaimDevice(uuid)
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
+
+		if rawJson {
+			output.PrintJsonOrError(resp)
+		} else {
+			printSuccess("device successfully unclaimed.")
+		}
+
+		return nil
+	},
+}
+
+var dashboardDeviceAttachCommand = &cli.Command{
+	Name:      "attach",
+	Usage:     "Attach self or another device to a group",
+	ArgsUsage: "[device uuid OR Husarnet IPv6 addr OR hostname] <group uuid OR name>",
+	Action: func(ctx context.Context, cmd *cli.Command) error {
+		requiredArgumentsRange(cmd, 1, 2)
+
+		// determine inputs
+		var paramDevice string
+		var paramGroup string
+		if cmd.Args().Len() == 2 {
+			paramDevice = cmd.Args().Get(0)
+			paramGroup = cmd.Args().Get(1)
+		} else {
+			status := getDaemonStatus()
+			// TODO: error here if we can't contact daemon
```

Comment:
Maybe add a flag to `getDaemonStatus` to indicate whether this call is strictly required and should print out a provided error message and quit if it's not available?